src/controller/api/thought.js

```diff
@@ -0,0 +1,150 @@
+const router = require("express").Router();
+
+const { Thought, User } = require("../../models/");
+
+module.exports = {
+  // Get all thoughts
+  async getThoughts(req, res) {
+    try {
+      const thoughts = await Thought.find({});
+      res.json({ thoughts });
+    } catch (error) {
+      console.log(error);
+      res
+        .status(500)
+        .send({ msg: "Something went wrong while retrieving thoughts." });
+    }
+  },
+  // Get single thought
+
+  async getSingleThought(req, res) {
+    try {
+      const thought = await Thought.findOne({ _id: req.params.thoughtId })
+        .select("-__v")
+        .populate({ path: "username", select: "-__v" });
+      if (!thought) {
+        res.status(404).json({ msg: "No thought with this ID." });
+      } else {
+        res.json(thought);
+      }
+    } catch (error) {
+      res
+        .status(500)
+        .send({ msg: "Something went wrong while retrieving the thought." });
+    }
+  },
+
+  // Create thought
+  async createThought(req, res) {
+    try {
+      const newThought = await Thought.create({
```

Comment:
Consider adding additional validation here e.g. only create the thought if the request contains all of the required fields from the Thought model.
This will prevent your app from breaking if people send requests with incorrect data.